src/main/java/io/strimzi/kafka/bridge/http/HttpAdminClientEndpoint.java

```diff
@@ -0,0 +1,132 @@
+
+            default:
+                throw new IllegalArgumentException("Unknown Operation: " + this.httpBridgeContext.getOpenApiOperation());
+
+        }
+    }
+
+    public void doListTopics(RoutingContext routingContext) {
+        listTopics(listTopicsResult -> {
+            if (listTopicsResult.succeeded()) {
+                JsonArray root = new JsonArray();
+                Set<String> topics = listTopicsResult.result();
+                topics.forEach(topic -> {
+                    root.add(topic);
+                });
+                HttpUtils.sendResponse(routingContext, HttpResponseStatus.OK.code(), BridgeContentType.KAFKA_JSON, root.toBuffer());
+            } else {
+                HttpBridgeError error = new HttpBridgeError(
+                        HttpResponseStatus.INTERNAL_SERVER_ERROR.code(),
+                        listTopicsResult.cause().getMessage()
+                );
+                HttpUtils.sendResponse(routingContext, HttpResponseStatus.INTERNAL_SERVER_ERROR.code(),
+                        BridgeContentType.KAFKA_JSON, error.toJson().toBuffer());
+            }
+        });
+    }
+
+    public void doGetTopic(RoutingContext routingContext) {
+        String topicName = routingContext.pathParam("topicname");
+        describeTopics(Arrays.asList(topicName), describeTopicsResult -> {
+            if (describeTopicsResult.succeeded()) {
+                Map<String, TopicDescription> topicDescriptions = describeTopicsResult.result();
+                JsonObject root = new JsonObject();
+                JsonArray partitionsArray = new JsonArray();
+                root.put("name", topicName);
+                TopicDescription description = topicDescriptions.values().iterator().next();
+                description.getPartitions().forEach(partitionInfo -> {
+                    int leaderId = partitionInfo.getLeader().getId();
+                    JsonObject partition = new JsonObject();
+                    partition.put("partition", partitionInfo.getPartition());
+                    partition.put("leader", leaderId);
+                    JsonArray replicasArray = new JsonArray();
+                    Set<Integer> insyncSet = new HashSet<Integer>();
+                    partitionInfo.getIsr().forEach(node -> {
+                        insyncSet.add(node.getId());
+                    });
+                    partitionInfo.getReplicas().forEach(node -> {
+                        JsonObject replica = new JsonObject();
+                        replica.put("broker", node.getId());
+                        replica.put("leader", leaderId == node.getId());
+                        replica.put("in_sync", insyncSet.contains(node.getId()));
+                        replicasArray.add(replica);
+                    });
+                    partition.put("replicas", replicasArray);
+                    partitionsArray.add(partition);
+                });
+                root.put("partitions", partitionsArray);
```

Comment:
I see the OpenAPI spec having a "configs" map with the topic configuration overrides. Where is it filled? You should be able to get this information using the `describeConfigs` method on the Vert.x Kafka admin client.

Reply:
ok. I added this call to fill the config.